tactics/equality.ml

```diff
@@ -331,34 +331,37 @@ let lib_ref_opt_pos name pos =
-    | false, _ , false , QConstant QType , true -> lib_ref_opt_pos "core.eq.rec_r" (AtPosition 5)
-    | false, _ , false , QConstant QProp , _ -> lib_ref_opt_pos "core.eq.ind_r" (AtPosition 5)
-    | false, _ , false , QConstant QSProp , _ -> lib_ref_opt_pos "core.eq.sind_r" (AtPosition 5)
-    | false, _ , true , QConstant QProp , _ -> lib_ref_opt_pos "core.eq.ind" (AtPosition 5)
-    | false, _ , true , QConstant QSProp , _ -> lib_ref_opt_pos "core.eq.sind" (AtPosition 5)
-    | false, _ , true , QConstant QType , false -> lib_ref_opt_pos "core.eq.rect" (AtPosition 5)
-    | false, _ , true , QConstant QType , true -> lib_ref_opt_pos "core.eq.rec" (AtPosition 5)
+    | false, Some true, true , QConstant QType , false -> Some ("rect_r", AtPosition 5)
+    | false, Some true, true , QConstant QType , true -> Some ("rec_r", AtPosition 5)
+    | false, Some true, true , QConstant QProp , _ -> Some ("ind_r",AtPosition 5)
+    | false, Some true, true , QConstant QSProp , _ -> Some ("sind_r",AtPosition 5)
+    | false, Some true, false , QConstant QType , false -> Some ("rect",AtPosition 5)
+    | false, Some true, false , QConstant QType , true -> Some ("rec",AtPosition 5)
+    | false, Some true, false , QConstant QProp , _ -> Some ("ind",AtPosition 5)
+    | false, Some true, false , QConstant QSProp , _ -> Some ("sind",AtPosition 5)
+    | false, _ , false , QConstant QType , false -> Some ("rect_r",AtPosition 5)
+    | false, _ , false , QConstant QType , true -> Some ("rec_r",AtPosition 5)
+    | false, _ , false , QConstant QProp , _ -> Some ("ind_r",AtPosition 5)
+    | false, _ , false , QConstant QSProp , _ -> Some ("sind_r",AtPosition 5)
+    | false, _ , true , QConstant QProp , _ -> Some ("ind",AtPosition 5)
+    | false, _ , true , QConstant QSProp , _ -> Some ("sind",AtPosition 5)
+    | false, _ , true , QConstant QType , false -> Some ("rect",AtPosition 5)
+    | false, _ , true , QConstant QType , true -> Some ("rec",AtPosition 5)
     (* Dependent case *)
-    | true, Some true, true , QConstant QType , _ -> lib_ref_opt_pos "core.eq.rect_r_dep" (AtPosition 5)
-    | true, Some true, true , QConstant QProp , _ -> lib_ref_opt_pos "core.eq.ind_r_dep" (AtPosition 5)
-    | true, _ , true , QConstant QType , _  -> lib_ref_opt_pos "core.eq.rect_dep" (AtPosition 5)
-    | true, _ , true , QConstant QProp , _  -> lib_ref_opt_pos "core.eq.ind_dep" (AtPosition 5)
+    | true, Some true, true , QConstant QType , _ -> Some ("rect_r_dep",AtPosition 5)
+    | true, Some true, true , QConstant QProp , _ -> Some ("ind_r_dep",AtPosition 5)
+    | true, _ , true , QConstant QType , _  -> Some ("rect_dep",AtPosition 5)
+    | true, _ , true , QConstant QProp , _  -> Some ("ind_dep",AtPosition 5)
     | _ , _, _ , _ , _ -> None
 
+let eq_scheme_name name dep lft2rgt inccl target is_set =
+  Option.bind (eq_scheme_pattern dep lft2rgt inccl target is_set)
+    (fun (recursor,pos) -> lib_ref_opt_pos ("core." ^ name ^ "." ^ recursor) (AtPosition 5))
```

Comment:
```suggestion
    (fun (recursor,pos) -> lib_ref_opt_pos ("core." ^ name ^ "." ^ recursor) pos)
```
or if the position is always 5 stop returning it from eq_scheme_pattern

Reply:
stop returning it from eq_scheme_pattern for now